src/control/cmd/daos/container.go

```diff
@@ -98,40 +111,97 @@ func (cmd *containerBaseCmd) closeContainer() error {
+	rc := C.daos_cont_query(cmd.cContHandle, &ci.dci, props, nil)
 	if err := daosError(rc); err != nil {
 		return nil, err
 	}
 
+	lType := C.get_dpe_val(&entries[0])
+	C.daos_unparse_ctype(C.ushort(lType), &cType[0])
+	ci.Type = C.GoString(&cType[0])
+
+	if lType == C.DAOS_PROP_CO_LAYOUT_POSIX {
+		var dfs *C.dfs_t
+		var attr C.dfs_attr_t
+		var oclass [10]C.char
+
+		rc := C.dfs_mount(cmd.cPoolHandle, cmd.cContHandle, C.O_RDONLY, &dfs)
+		if err := dfsError(rc); err != nil {
+			return nil, errors.Wrap(err, "failed to mount container")
+		}
+
+		rc = C.dfs_query(dfs, &attr)
+		if err := dfsError(rc); err != nil {
+			return nil, errors.Wrap(err, "failed to query container")
+		}
+		C.daos_oclass_id2name(attr.da_oclass_id, &oclass[0])
+		ci.ObjectClass = C.GoString(&oclass[0])
+		ci.ChunkSize = uint64(attr.da_chunk_size)
+
+		if err := dfsError(C.dfs_umount(dfs)); err != nil {
+			return nil, errors.Wrap(err, "failed to unmount container")
+		}
+	}
+
 	return ci, nil
 }
 
+func (cmd *containerBaseCmd) connectPool(ap *C.struct_cmd_args_s) (func(), error) {
+	if err := cmd.poolBaseCmd.connectPool(); err != nil {
+		return nil, err
+	}
+
+	if ap != nil {
+		ap.pool = cmd.cPoolHandle
+		if err := copyUUID(&ap.p_uuid, cmd.poolUUID); err != nil {
+			cmd.disconnectPool()
+			return nil, err
+		}
+	}
+
+	return cmd.disconnectPool, nil
+}
+
 type containerCreateCmd struct {
 	containerBaseCmd
 
-	UUID        string         `long:"cont" short:"c" description:"container UUID (optional)"`
-	Type        string         `long:"type" short:"t" description:"container type" choice:"POSIX" choice:"HDF5" default:"POSIX"`
+	Type        string         `long:"type" short:"t" description:"container type" choice:"POSIX" choice:"HDF5"`
 	Path        string         `long:"path" short:"d" description:"container namespace path"`
-	ChunkSize   chunkSizeFlag  `long:"chunk-size" short:"z" description:"container chunk size"`
-	ObjectClass objClassFlag   `long:"oclass" short:"o" description:"default object class"`
+	ChunkSize   ChunkSizeFlag  `long:"chunk-size" short:"z" description:"container chunk size"`
+	ObjectClass ObjClassFlag   `long:"oclass" short:"o" description:"default object class"`
 	Properties  PropertiesFlag `long:"properties" description:"container properties"`
-	Mode        consModeFlag   `long:"mode" short:"M" description:"DFS consistency mode"`
+	Mode        ConsModeFlag   `long:"mode" short:"M" description:"DFS consistency mode"`
 	ACLFile     string         `long:"acl-file" short:"A" description:"input file containing ACL"`
 	User        string         `long:"user" short:"u" description:"user who will own the container (username@[domain])"`
 	Group       string         `long:"group" short:"g" description:"group who will own the container (group@[domain])"`
+	ContFlag    ContainerID    `long:"cont" short:"c" description:"container UUID (optional)"`
+	Args        struct {
+		Container ContainerID `positional-arg-name:"<container UUID (optional)>"`
```

Comment:
Should it be "container name or UUID" ?

Reply:
Yep. Will fix this and make sure it's consistent with the pool help strings.

Reply:
Oh. Actually, this is the create command, and this argument is for optionally supplying the container UUID to be used on create. There has been some discussion about removing this for the same reasons we removed it from the pool create flow, but for the moment we have to support it.

Reply:
On this topic I've got another proposal to be able to pick up the pool from the path, so container create could work without a pool uuid/label set on the command line which will impact this code I suspect.

https://github.com/daos-stack/daos/pull/5969

Reply:
Had a look at that PR. Will be easier to integrate that feature into the new tool after this one lands, as it cleans up a bunch of stuff.

Reply:
I thought so, that's why I've paused it but I didn't want you to add something now that would make it more difficult later.